src/app/sensor-detail/sensor-detail.service.test.js

```diff
@@ -0,0 +1,44 @@
+import SensorDetailService from './sensor-detail.service';
+import {Http} from 'angular2/http';
+import {beforeEachProviders} from 'angular2/testing';
+import {provide} from 'angular2/core';
+
+const observableMock = { map() {} };
+
+class HttpMock {
+    get() { return observableMock; }
+    put() { return observableMock; }
+}
+
+describe('SensorDetailService', () => {
+    let sut;
+    let httpMock;
+
+    beforeEachProviders(() => [
+        provide(Http, {useClass: HttpMock})
+    ]);
+
+    beforeEach(() => {
+        httpMock = new HttpMock();
+        spyOn(httpMock, 'get').and.callThrough();
+        spyOn(httpMock, 'put').and.callThrough();
+        sut = new SensorDetailService(httpMock);
+    });
+
+    it('should be defined', () => {
+        expect(sut).toBeDefined();
+    });
+
+    it('should get sensor data from the server', () => {
+        const idMock = 'mock';
+        sut.get(idMock);
+        expect(httpMock.get).toHaveBeenCalledWith(`/sensors/${idMock}`);
+    });
+
+    it('should save sensor', () => {
+        const sensorMock = {_id: 'mock'};
+        const body = JSON.stringify(sensorMock);
```

Comment:
oops, missed
was actually addressing line 16 in sensor-detail.service.js
`const body = JSON.stringify(sensor);`